README.md

```diff
@@ -1,19 +1,83 @@
 
-Note #1: The connector framework is a tech preview feature. Tech preview features
-are subject to change and are not covered by the support SLA of general release
-(GA) features. Elastic plans to promote this feature to GA in a future release.
+[//]: # (The introduction should:)
+[//]: # (- Summarize the tools provided in this repository.)
+[//]: # (- Summarize the procedures and reference information from this readme and any other docs included in this repo.)
+[//]: # (- Introduce any terminology that is our own.)
+[//]: # (- Identify this feature as a technical preview.)
+[//]: # (Then present a TOC to guide the user through the following sections.)
 
-Note #2: The `main` branch of this repository is currently being heavily reworked. If you want to use the
-connector clients now, it would be advisable to check out/fork the [latest stable branch](https://github.com/elastic/connectors-ruby/tree/8.3).
+Note #1: The connector framework is a tech preview feature. Tech preview features are subject to change and are not covered by the support SLA of general release (GA) features. Elastic plans to promote this feature to GA in a future release.
 
-### System Requirements
+## Terminology
+
+* `connector` - generic term used to refer to both native connectors and connector clients.
+* `connector client` - specific light-weight connector implementation, open-code. Not all clients will be supported as native, but we target to have native connectors also offered as clients. Connector clients can be built by Elastic or community built.
+
+## Disambiguation: connectors-ruby and connectors-python
+
+This repository contains connector clients written in Ruby. However, some connectors will also be implemented in Python. The Python connectors are located in a separate repository, [elastic/connectors-python](https://github.com/elastic/connectors-python).
+
+Before getting started, review important information about this feature:
+
+- [Known issues and limitations](#known-issues-and-limitations)
+- [Getting help and providing feedback](#getting-help-and-providing-feedback)
+
+Build, deploy, and operate your connector:
+
+- [Building a connector](#building-a-connector)
+- [Moving to production](#moving-to-production)
+- [Publishing a connector (optional)](#publishing-a-connector-optional)
+
+Reference:
+
+- [Connector protocol reference](#connector-protocol-reference)
+
+## Known issues and limitations
+
+[//]: # (If the only "limitation" is technical preview, we could drop this section.)
+- The connector clients implemented in this repository are [mongodb](/lib/connectors/mongodb), [gitlab](/lib/connectors/gitlab) and [stub connector](lib/connectors/stub_connector). None of those are production ready clients - rather, they are functional examples, which you can use as a base for your own work.
```

Comment:
I remember @artem-shelkovnikov mentioning something about a scheduling / sync limitation as well. He wanted to create a ticket, but I don't have one as reference.